addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4222,7 +4255,11 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
             } else {
                 shapeToSelect = possibleShapes[1];
             }
-            return this._handlePreviewState(false, () => ({shape: shapeToSelect}));
+            setTimeout(() => {
+                // options for shape will only be available after _toggleShape() returned
+                this._requestUserValueWidgets('bg_shape_opt')[0].$el.find('we-toggler')[0].click();
+            }, 1);
```

Comment:
Seems like the review lost one of my comments: `setTimeout` with 0ms (or simply omitting the time parameter) will work fine: it will create an event that will not be processed until the current event (the click on the shape button) has been processed. All of this is assuming that this is synchronous however, so in this case I think using `snippet_edition_request` might be safer.